query-engine/driver-adapters/executor/src/wasm.ts

```diff
@@ -1,13 +1,35 @@
-import * as wasm from '../../../query-engine-wasm/pkg/query_engine_bg.js'
+import * as wasmPostgres from '../../../query-engine-wasm/pkg/postgresql/query_engine_bg.js'
+import * as wasmMysql from '../../../query-engine-wasm/pkg/mysql/query_engine_bg.js'
+import * as wasmSqlite from '../../../query-engine-wasm/pkg/sqlite/query_engine_bg.js'
 import fs from 'node:fs/promises'
 import path from 'node:path'
 import { fileURLToPath } from 'node:url'
 
 const dirname = path.dirname(fileURLToPath(import.meta.url))
 
-const bytes = await fs.readFile(path.resolve(dirname, '..', '..', '..', 'query-engine-wasm', 'pkg', 'query_engine_bg.wasm'))
-const module = new WebAssembly.Module(bytes) 
-const instance = new WebAssembly.Instance(module, { './query_engine_bg.js': wasm })
-wasm.__wbg_set_wasm(instance.exports);
+const wasm = {
+    postgres: wasmPostgres,
+    mysql: wasmMysql,
+    sqlite: wasmSqlite
+}
 
-export const WasmQueryEngine = wasm.QueryEngine
\ No newline at end of file
+type EngineName = keyof typeof wasm;
+
+const initializedModules = new Set<EngineName>()
+
+
+
+export async function getEngineForProvider(provider: EngineName) {
+    const engine = wasm[provider]
+    if (!initializedModules.has(provider)) {
+        const subDir = provider === 'postgres' ? 'postgresql' : provider
```

Comment:
I'm in favor of consolidating the naming of the quaint feature flag in alignment with the rest of the codebase in accordance with your comment https://github.com/prisma/prisma-engines/pull/4681/files#r1474832004

Reply:
ok, i had a look at the engines codebase and:
- `feature="postgresql"` is used 247 times, most of them in quaint.
- string "postgresql" is encountered 665 times in general

So, if anything, `postgres` seems like an outlier. In any case, I feel like changing all of them is too much for this PR. Would you agree?

Reply:
Yes, let's postpone it. https://github.com/prisma/team-orm/issues/925